### ISSUE TITLE
Prośba o aktywację konta

### ISSUE THREAD

**Issue by Dvwid** [allegro/allegro-api#9213]
Proszę o aktywację konta gumocze ponieważ nie przechodzi szybka weryfikacja płatności.

**Comment by MaciejFrackowiak** [comment]
Konto aktywowane.

**Comment by Qmiro** [comment]
Proszę o aktywację konta Client:107190111 na sandboxie

**Comment by AureliuszBrussy** [comment]
@Qmiro to jest konto zwykłe, aktywujesz je przy pierwszej testowej transakcji. Jeżeli chcesz testować fukcjonalności związane ze sprzedażą - [zarejestruj nowe konto firmowe](https://allegro.pl.allegrosandbox.pl/rejestracja/konto-firmowe/nowe-konto?origin_url=%2F).

**Comment by Qmiro** [comment]
Proszę o aktywację nowego konta firmowego mkpnankiety

**Comment by AureliuszBrussy** [comment]
@Qmiro aktywowałem Twoje konto.

**Comment by Qmiro** [comment]
Dodałęm aplikacje do api i przy próbie wywołania endpointa ttps://api.allegro.pl.allegrosandbox.pl/sale/offers dostaje 403 (No access to the specified resource.) mimo że mam dodane uprawneinie allegro:api:sale:offers:read np dla endpointa sale/categories zwracana jest liczba kateggori.

**Comment by AureliuszBrussy** [comment]
Nie korzystasz przypadkiem z autoryzacji typu [Client_credentials flow](https://developer.allegro.pl/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR#clientcredentials-flow), która nie ma kontekstu użytkownika, a jedynie kontekst aplikacji? Aby posiadać access token z kontekstem użytkownika niezbędne jest użycie [Authorization Code flow](https://developer.allegro.pl/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR#authorization-code-flow) lub [Device flow](https://developer.allegro.pl/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR#device-flow).

**Comment by Qmiro** [comment]
Wybierając zgodnie z opisem  [Device flow](https://developer.allegro.pl/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR#device-flow) - umożliwia autoryzację na urządzeniach lub w aplikacjach, które nie posiadają interfejsu graficznego lub sposobu na wpisanie tekstu, Tylko po wygenerowaniu kodu device_code trzeba potwierdzić link verification_uri_complete czyli defakto musi być dostęp do interfejsu graficznego. Czy można uzyskać dostep w aplikacji konsolowej bez tego potwierdzenia w graficznym interfejsie ?

**Comment by AureliuszBrussy** [comment]
Trzeba potwierdzić link, ale można zrobić to na innym urządzeniu czy innym środowisku. W procesie OAuth nie uda się pominąć interakcji użytkownik, musi on w którymś momencie wyrazić zgodę na działanie aplikacji w jego imieniu. Dodam, że konieczna jest pierwsza interakcja - później wystarczy [odświeżać](https://developer.allegro.pl/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR#przedluzenie-waznosci-tokena) uzyskany access token za pomocą refresh tokena.

**Comment by Qmiro** [comment]
Aplikacja będzie działać w środowisku bez dostępu do przeglądarki albo klawiatury (np. aplikacja konsolowa albo na urządzeniu typu telewizor) aplikacja konsolowa - nie ma dostępu do tego środowiska graficznego. Czy uda się potwierdzić tożsamość użytkownika bez klikania w urządzeniu (cron konsoli nie ma urządzenia) Cały proces musi się odbywać w konsoli bez udziału fizycznie człowieka (przez przesyłanie danych w celu uzyskania pełnej autoryzacji)

**Comment by AureliuszBrussy** [comment]
Tak jak wspomniałem - nie ma takiej możliwości. Natomiast w ramach device flow możesz przekazać verification_uri_complete użytkownikowi, a ten może otworzyć go samodzielnie np. w przeglądarce. Wtedy z poziomu przeglądarki wyraża zgodę na działanie aplikacji w jego imieniu.

**Comment by Qmiro** [comment]
"Wtedy z poziomu przeglądarki wyraża zgodę na działanie aplikacji w jego imieniu." - Mimo wszystko nawet żeby potwierdzić ten link i tak się musi zalogować.

**Comment by AureliuszBrussy** [comment]
Oczywiście, natomiast na innym urządzeniu lub środowisku. Tak jak wspomniałem - użytkownik musi wyrazić zgodę na działanie aplikacji w jego imieniu, więc wyświetlenie tzw. consent screena przynajmniej przy pierwszym skorzystaniu z aplikacji jest konieczne. Nie ominiesz tego. Po wygenrowaniu pierwszej pary tokenów, będziesz mógł użyć refresh token i kolejne logowanie nie będzie konieczne.
